xgi/linalg/hypergraph_matrix.py

```diff
@@ -0,0 +1,289 @@
+"""Matrices associated to hypergraphs.
+
```

Comment:
I think that for now, having the same header for every file might be okay, but we should at least change the first line because that shows up when navigating through readthedocs.

Reply:
I updated the first lines to be a bit more specific